### PR TITLE
plugins: fix cmake plugin prefix

### PR DESCRIPTION
**Pull request by cmatsuoka**
Properly format the cmake prefix path string, and set the installation
prefix to match v1 plugins.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----
CRAFT-1154

### CODE REVIEW

File: craft_parts/plugins/cmake_plugin.py [hunk 1]
```diff
@@ -88,9 +88,7 @@ def get_build_packages(self) -> Set[str]:
 
     def get_build_environment(self) -> Dict[str, str]:
         """Return a dictionary with the environment to use in the build step."""
-        return {
-            "CMAKE_PREFIX_PATH": "{self._part_info.stage_dir}",
-        }
+        return {"CMAKE_PREFIX_PATH": f"{self._part_info.stage_dir}"}
```
Comment (sergiusens) [inline craft_parts/plugins/cmake_plugin.py:91]: not related to the PR, but why is this set to stage_dir?

Comment (sergiusens) [inline craft_parts/plugins/cmake_plugin.py:91]: Side question, is this a better alternative?
```suggestion
        return {"CMAKE_PREFIX_PATH": str(self._part_info.stage_dir)}
```

File: craft_parts/plugins/cmake_plugin.py [hunk 2]
```diff
@@ -101,6 +99,7 @@ def get_build_commands(self) -> List[str]:
             f'"{self._part_info.part_src_subdir}"',
             "-G",
             f'"{options.cmake_generator}"',
+            "-DCMAKE_INSTALL_PREFIX=",
```
Comment (sergiusens) [inline craft_parts/plugins/cmake_plugin.py:102]: please add a code comment on the effects or why this is here.

File: craft_parts/plugins/cmake_plugin.py (hunk 1 above)
Comment (cmatsuoka) [reply craft_parts/plugins/cmake_plugin.py:91]: This allows cmake to find staged headers and libs. The regular search paths for system headers and libraries are still used, this only adds staged locations to the list.

Comment (cmatsuoka) [reply craft_parts/plugins/cmake_plugin.py:91]: It is, using that instead.

File: craft_parts/plugins/cmake_plugin.py (hunk 2 above)
Comment (cmatsuoka) [reply craft_parts/plugins/cmake_plugin.py:102]: Comment added for this variable and for `CMAKE_PREFIX_PATH` above.

**Comment by cmatsuoka** [comment]
Amended and rebased on main.

**Review by mr-cal — APPROVED** [review]
Looks good.  I tested with a project that previously had 
```
cmake-parameters:
  - -DCMAKE_INSTALL_PREFIX=/usr/
  - -DCMAKE_PREFIX_PATH="${SNAPCRAFT_STAGE}/usr"
```
in a few of the parts.  Using this PR, that snap builds with or without those extra parameters.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**